src/Nethermind/Nethermind.Blockchain.Test/TransactionFilterTests.cs

```diff
@@ -0,0 +1,101 @@
+            _filter = new AcceptAllTransactionFilter();
+            var transactions = GetTransactions();
+            var addedTransactions = ApplyCanAddFilter(transactions);
+            var deletedTransactions = ApplyCanDeleteFilter(transactions);
+            addedTransactions.Length.Should().Be(transactions.Length);
+            deletedTransactions.Length.Should().Be(transactions.Length);
+        }
+
+        [Test]
+        public void should_not_accept_any_transaction_when_using_reject_all_filter()
+        {
+            _filter = new RejectAllTransactionFilter();
+            var transactions = GetTransactions();
+            var addedTransactions = ApplyCanAddFilter(transactions);
+            var deletedTransactions = ApplyCanDeleteFilter(transactions);
+            addedTransactions.Should().BeEmpty();
+            deletedTransactions.Should().BeEmpty();
+        }
+
+        [Test]
+        public void should_add_some_transactions_to_storage_when_using_accept_when_filter()
+        {
+            _filter = AcceptWhenTransactionFilter
+                .Create()
+                .AddWhen()
+                .Nonce(n => n >= 0)
+                .GasPrice(p => p > 2 && p < 1500)
+                .Build()
+                .DeleteWhen()
+                .GasLimit(l => l < 1000)
+                .Build()
+                .BuildFilter();
+
+            var transactions = GetTransactions();
+            var addedTransactions = ApplyCanAddFilter(transactions);
+            var deletedTransactions = ApplyCanDeleteFilter(transactions);
+            addedTransactions.Should().NotBeEmpty();
+            deletedTransactions.Should().NotBeEmpty();
+        }
+
+        private Transaction[] ApplyCanAddFilter(IEnumerable<Transaction> transactions)
+            => transactions.Where(t => _filter.CanAdd(t)).ToArray();
+
+        private Transaction[] ApplyCanDeleteFilter(IEnumerable<Transaction> transactions)
+            => transactions.Where(t => _filter.CanDelete(t)).ToArray();
+
+        private Transaction[] GetTransactions()
+            => new[]
+            {
+                GetTransaction(0, 1000, 10, Address.Zero, new byte[0], TestObject.PrivateKeyA),
+                GetTransaction(0, 500, 2, Address.FromNumber(1), new byte[0], TestObject.PrivateKeyB),
+                GetTransaction(1, 2000, 50, Address.FromNumber(3), new byte[0], TestObject.PrivateKeyC),
+                GetTransaction(2, 10000, 100, Address.FromNumber(2), new byte[0], TestObject.PrivateKeyD),
+                GetTransaction(3, 4000, 5, Address.Zero, new byte[0], TestObject.PrivateKeyC)
+            };
+
+        private Transaction GetTransaction(UInt256 nonce, UInt256 gasLimit,
+            UInt256 gasPrice, Address to, byte[] data, PrivateKey privateKey)
+            => Build.A.Transaction
+                .WithNonce(nonce)
+                .WithGasLimit(gasLimit)
+                .WithGasPrice(gasPrice)
+                .WithData(data)
+                .To(to)
```

Comment:
should we follow conventions and call it WithTo?

Reply:
I thought about it  - `WithTo()` sticks to the convention, however it didn't make much sens to me as a phrase, but I can change it.

Reply:
totally understand, it is tempting to keep it, you can add both!

Reply:
Ok